src/plugins/image_embeddable/public/components/image_embeddable.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { ReactEmbeddableDynamicActionsApi } from '@kbn/embeddable-enhanced-plugin/public/plugin';
+import { PublishingSubject, useStateFromPublishingSubject } from '@kbn/presentation-publishing';
+
+import { imageClickTrigger } from '../actions';
+import { ImageEmbeddableStrings } from '../image_embeddable/image_embeddable_strings';
+import { ImageEmbeddableApi } from '../image_embeddable/types';
+import { FileImageMetadata, FilesClient, imageEmbeddableFileKind } from '../imports';
+import { coreServices, screenshotModeService, uiActionsService } from '../services/kibana_services';
+import { ImageConfig } from '../types';
+import { createValidateUrl } from '../utils/validate_url';
+import { ImageViewer } from './image_viewer';
+import { ImageViewerContext } from './image_viewer/image_viewer_context';
+
+import './image_embeddable.scss';
+
+interface ImageEmbeddableProps {
+  api: ImageEmbeddableApi & {
+    setBlockingError: (error: Error | undefined) => void;
+    setDataLoading: (loading: boolean | undefined) => void;
```

Comment:
The setters for the publishing subjects aren't part of `PublishesDataLoading` + `PublishesBlockingError` and I wasn't sure if they should be...  So, I just added them here to keep the props clean.